lib/logstash/outputs/elasticsearch/http_client_builder.rb

```diff
@@ -197,5 +197,14 @@ def self.setup_api_key(logger, params)
     def self.dedup_slashes(url)
       url.gsub(/\/+/, "/")
     end
+
+    def self.resolve_filter_path(url)
+      return url if url.nil? || url.match?(/(?:[&|?])filter_path=/)
```

Comment:
dedup_slashes can never return nil since it performs gsub on a string without checking it it's nil:

```suggestion
      return url if url.match?(/(?:[&|?])filter_path=/)
```